python/lsst/daf/butler/_butler.py

```diff
@@ -2037,16 +2044,17 @@ def import_(
 
         Parameters
         ----------
-        directory : `str`, optional
+        directory : `str` or `~lsst.resources.ResourcePath`, optional
             Directory containing dataset files to import from. If `None`,
             ``filename`` and all dataset file paths specified therein must
             be absolute.
-        filename : `str` or `TextIO`, optional
+        filename : `~lsst.resources.ResourcePathExpression` or `TextIO`
             A stream or name of file that contains database information
             associated with the exported datasets, typically generated by
-            `~lsst.daf.butler.Butler.export`.  If this a string (name) and
-            is not an absolute path, does not exist in the current working
-            directory, and ``directory`` is not `None`, it is assumed to be in
+            `~lsst.daf.butler.Butler.export`.  If this a string (name) or
+            `~lsst.resources.ResourcePath` and is not an absolute path,
+            does not exist in the current working directory, and ``directory``
+            is not `None`, it is assumed to be in
             ``directory``. Defaults to "export.{format}".
```

Comment:
Just a general though, not related to the changes - this handling of the `directory` resembles a search path which always includes two entries - `cwd` and `directory`. This may cause confusion if an export file can be found in both places but user really wants it to come from `directory`, as there is no way to remove `cwd` from a search path. Would it make sense to reverse search path to look in `directory` first, or have an option to disable search in `cwd`?

Reply:
I think @TallJimbo designed this API. I'm happy to change it to prefer the given directory over cwd if he agrees.

Reply:
I have no problem changing it.  Andy's argument makes sense.

Reply:
I've changed it to use directory if both cwd and directory have the file. I also issue a warning message in that scenario just in case.